disnake/ext/commands/params.py

```diff
@@ -911,7 +911,11 @@ def isolate_self(
         parametersl.pop(0)
     if parametersl:
         annot = parametersl[0].annotation
-        if issubclass_(annot, ApplicationCommandInteraction) or annot is inspect.Parameter.empty:
+        if (
+            issubclass_(annot, ApplicationCommandInteraction)
+            or annot is inspect.Parameter.empty
+            or issubclass_(get_origin(annot), Interaction)
+        ):
```

Comment:
this can be streamlined a bit:
```suggestion
        annot = parametersl[0].annotation
        annot = get_origin(annot) or annot
        if issubclass_(annot, ApplicationCommandInteraction) or annot is inspect.Parameter.empty:
```

Another thing - there are two other locations that should also call `get_origin` beforehand in a similar way:
https://github.com/DisnakeDev/disnake/blob/a309929c9b7c43eba5fa6103571d3fa562cb6def/disnake/ext/commands/params.py#L985 https://github.com/DisnakeDev/disnake/blob/a309929c9b7c43eba5fa6103571d3fa562cb6def/disnake/ext/commands/params.py#L1119